### PR TITLE
Bug/1423 get v2 servicepath long

### LINKED ISSUE #1423: [BUG] with some service-path header value, nothing is returned from db
### Datasets:
```
      | /serv1/serv2/serv3/serv4/serv5/serv6/serv7/serv8/serv9/serv10 |
      | max length allowed                                            | where max length = 50 chars
      | max length allowed and ten levels                             |
```

##### Create request
```
 POST http://localhost:1026/v2/entities
 Content-Type: application/json
 Fiware-Service: test_update_service_path
 Fiware-ServicePath: /EhZa1vtv49pfnQa8ffF_N0Bf4Er7skJA6nvA5tSQzsxCyC6Fv_/Gj1FrYs09OED9Sz4ECimZI1JCITymjv2YYXZBSK3WDjStdcppq/HBXz8BmEJ3RZNH2XIyy9_OnKU8LAmHw2DByAln6DaxgKSkt3HC/rIuun2LhD_Ed4mf98XHU2_V0Pe2RGRsjyP9iAXHv7cuywSdxnT/nJf4oSd9tynFa_IwIEZSqvC6BEeup_qeZyfFaXAD5oO5Yx13Gj/9SASSborWAojYtX0Y10Oy0_2uITVnJglWNXj9svGDyrh1H1iM8/b1zRgnH378rW3jcp7mRe5TQfY_WZvvqMNQud0B8yjjt8EJwbsN/nl1dEI0ApCf1f0YBMOl_jj6uO5P2wAwaliIBu8eul1ORE4Xed7/4IQmeTFEeXv0GDxyZE9TIPWmk3vUFiEkd4kz0XHfUWbbOQSjGG/n5BcG296qVRG8x1z3SgG2usNcvednOyfM1BmFbm89zU3SgRKMj
 {"temperature_2": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "temperature_0": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "temperature_1": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "type": "house", "id": "room"}
```

##### docs in mongodb
```
use orion-test_update_service_path
> db.entities.find()
{ "_id" : { "id" : "room", "type" : "house", "servicePath" : "/serv1/serv2/serv3/serv4/serv5/serv6/serv7/serv8/serv9/serv10" }, "attrNames" : [ "temperature_2", "temperature_0", "temperature_1" ], "attrs" : { "temperature_2" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_0" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_1" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] } }, "creDate" : 1445954322, "modDate" : 1445954322 }
{ "_id" : { "id" : "room", "type" : "house", "servicePath" : "/PKcFuHi5bqkpHysogtmXaxt7Vh2Q2ehOfLfwv9RQ7xi1S3_nJ3" }, "attrNames" : [ "temperature_2", "temperature_0", "temperature_1" ], "attrs" : { "temperature_2" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_0" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_1" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] } }, "creDate" : 1445954322, "modDate" : 1445954322 }
{ "_id" : { "id" : "room", "type" : "house", "servicePath" : "/EhZa1vtv49pfnQa8ffF_N0Bf4Er7skJA6nvA5tSQzsxCyC6Fv_/Gj1FrYs09OED9Sz4ECimZI1JCITymjv2YYXZBSK3WDjStdcppq/HBXz8BmEJ3RZNH2XIyy9_OnKU8LAmHw2DByAln6DaxgKSkt3HC/rIuun2LhD_Ed4mf98XHU2_V0Pe2RGRsjyP9iAXHv7cuywSdxnT/nJf4oSd9tynFa_IwIEZSqvC6BEeup_qeZyfFaXAD5oO5Yx13Gj/9SASSborWAojYtX0Y10Oy0_2uITVnJglWNXj9svGDyrh1H1iM8/b1zRgnH378rW3jcp7mRe5TQfY_WZvvqMNQud0B8yjjt8EJwbsN/nl1dEI0ApCf1f0YBMOl_jj6uO5P2wAwaliIBu8eul1ORE4Xed7/4IQmeTFEeXv0GDxyZE9TIPWmk3vUFiEkd4kz0XHfUWbbOQSjGG/n5BcG296qVRG8x1z3SgG2usNcvednOyfM1BmFbm89zU3SgRKMj" }, "attrNames" : [ "temperature_2", "temperature_0", "temperature_1" ], "attrs" : { "temperature_2" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_0" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] }, "temperature_1" : { "type" : "celcius", "creDate" : 1445954322, "modDate" : 1445954322, "value" : "34", "md" : [ { "name" : "very_hot_1", "type" : "alarm", "value" : "hot" }, { "name" : "very_hot_0", "type" : "alarm", "value" : "hot" } ] } }, "creDate" : 1445954322, "modDate" : 1445954322 }
```

##### Update request
```
PUT http://localhost:1026/v2/entities/room/attrs/temperature_0
Content-Type: application/json
Fiware-Service: test_update_service_path
Fiware-ServicePath: /EhZa1vtv49pfnQa8ffF_N0Bf4Er7skJA6nvA5tSQzsxCyC6Fv_/Gj1FrYs09OED9Sz4ECimZI1JCITymjv2YYXZBSK3WDjStdcppq/HBXz8BmEJ3RZNH2XIyy9_OnKU8LAmHw2DByAln6DaxgKSkt3HC/rIuun2LhD_Ed4mf98XHU2_V0Pe2RGRsjyP9iAXHv7cuywSdxnT/nJf4oSd9tynFa_IwIEZSqvC6BEeup_qeZyfFaXAD5oO5Yx13Gj/9SASSborWAojYtX0Y10Oy0_2uITVnJglWNXj9svGDyrh1H1iM8/b1zRgnH378rW3jcp7mRe5TQfY_WZvvqMNQud0B8yjjt8EJwbsN/nl1dEI0ApCf1f0YBMOl_jj6uO5P2wAwaliIBu8eul1ORE4Xed7/4IQmeTFEeXv0GDxyZE9TIPWmk3vUFiEkd4kz0XHfUWbbOQSjGG/n5BcG296qVRG8x1z3SgG2usNcvednOyfM1BmFbm89zU3SgRKMj
{"value": "80", "type": "Fahrenheit", "very_hot_2": {"type": "alarm", "value": "cold"}, "very_hot_1": {"type": "alarm", "value": "cold"}, "very_hot_0": {"type": "alarm", "value": "cold"}}
```

##### Update response
```
http code: 404
content-length: 0
```
----------------------------------------------------------------------------------------------------
##### Create request (5 request with id = room2_* (0-4))
```
 POST http://localhost:1026/v2/entities
 Content-Type: application/json
 Fiware-Service: test_service_path
 Fiware-ServicePath: /F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX
 {"type": "room", "temperature": "34", "id": "room2_4"}
```

##### docs in mongodb
```
use orion-test_service_path
> db.entities.find()
{ "_id" : { "id" : "room2_0", "type" : "room", "servicePath" : "/F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX" }, "attrNames" : [ "temperature" ], "attrs" : { "temperature" : { "type" : "", "creDate" : 1445955227, "modDate" : 1445955227, "value" : "34" } }, "creDate" : 1445955227, "modDate" : 1445955227 }
{ "_id" : { "id" : "room2_1", "type" : "room", "servicePath" : "/F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX" }, "attrNames" : [ "temperature" ], "attrs" : { "temperature" : { "type" : "", "creDate" : 1445955227, "modDate" : 1445955227, "value" : "34" } }, "creDate" : 1445955227, "modDate" : 1445955227 }
{ "_id" : { "id" : "room2_2", "type" : "room", "servicePath" : "/F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX" }, "attrNames" : [ "temperature" ], "attrs" : { "temperature" : { "type" : "", "creDate" : 1445955227, "modDate" : 1445955227, "value" : "34" } }, "creDate" : 1445955227, "modDate" : 1445955227 }
{ "_id" : { "id" : "room2_3", "type" : "room", "servicePath" : "/F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX" }, "attrNames" : [ "temperature" ], "attrs" : { "temperature" : { "type" : "", "creDate" : 1445955227, "modDate" : 1445955227, "value" : "34" } }, "creDate" : 1445955227, "modDate" : 1445955227 }
{ "_id" : { "id" : "room2_4", "type" : "room", "servicePath" : "/F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX" }, "attrNames" : [ "temperature" ], "attrs" : { "temperature" : { "type" : "", "creDate" : 1445955227, "modDate" : 1445955227, "value" : "34" } }, "creDate" : 1445955227, "modDate" : 1445955227 }
```

##### Get request
```
GET http://localhost:1026/v2/entities
Content-Type: application/json
Fiware-Service: test_service_path
Fiware-ServicePath: /F_n_F4elhADVuCbEK4DDXlmi8lcV1b81OIhOpiubSD2jO6BZGM/9WNRpTC7Citc4c3qPbrHBUB7YhMxr1jenFM3JMhBHshBT5PcKl/2UAwWWAKBHI_0zU7sLfjozmcyhEcVle2c6mHQFZFkvswiyH2nb/6QaTP24vcPf4GozreIJnqZUN7WHYoOoCLz2dcgdVUOJ3sdiIh3/uf8uPlJanFCgkSiYnPwzTbIMc637zTK2ND_MsBK9BwMcEbQ9Xr/au_RXEqbjEMNRIelgPjosLpV_ZZ_JZqBg0UC1L8wlbsgYPLQEc/knywh8I1Ke41Cs1rcT4AZfoI0F6EsGyuVcgJTP3VJRqpRfTv7j/HTbaBPgyw4Z94_kzb40tCMzAWzYnerChJQmMeGND_kX6zXnBAO/QjyQLjX3OPyApIg0s4nwAqFGIoGmytUVOt2k4o0qpWNqvwxufg/RtEgL9yEQBp1wSX4bYwze7kf_BMh0sISAJBPNFnH5c9r6ydxsX
```

##### Get response
```
 http code: 200
 content-length: 2
 content-type: application/json
 []
```

##### Create request
```
  POST http://localhost:1026/v2/entities
  Content-Type: application/json
  Fiware-Service: test_delete_service_path
  Fiware-ServicePath: /UK6AMvm_Rz2yA_vR_HFsf0TAM2BLuQalyHZ_xqANkHEG_yKujr/BOkaWamQeL3JYQjN1NUiy_S3J3kw246N_MEmZ0zmmHGj8wP9Ww/no8GoBTJooh2Gv17tyOS1MYXKt0nnkDXRCSYhIMEE4p8iwRfdT/4Wke_KdxtJdRwIk7PMigRKNsaP_Y7ZPrwuaU4CBeGnTHcYR_pl/UxQQ2Bxyo39xPU9rVpV4oDbLWiOqvDxPUavwMFRFp7SYjlpJlT/3RAAU_jqgU4RxR5O4fJ4ij9Y3VrG3YfH2omuhyzcYErjFJ4mCN/b4TAvJUt1BemF4FMwHlb_33TT6GSWeKY_a5fbOm8EZ2f9KZo9t/QT9jAX8IpPhQeBS942ANt8txdypvkHwvZfEeuZdkb0WsJEoael/rTXDwCrXIh6QpLsqVziFdU6ioQ8Y6QxfH0Qt6YuIqdZIE87QWH/cgtk7TSxNu1XTBmMALsoMabFuUaEhgji45IAXtmciSln9x_HPx
 {"temperature_2": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "temperature_0": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "temperature_1": {"type": "celcius", "value": "34", "very_hot_1": {"type": "alarm", "value": "hot"}, "very_hot_0": {"type": "alarm", "value": "hot"}}, "type": "house", "id": "room"}
```

##### docs in mongodb
```
use orion-test_update_service_path
> db.entities.find()
 [{"modDate": 1447667724, "attrNames": ["temperature_2", "temperature_0", "temperature_1"], "_id": {"type": "house", "id": "room", "servicePath": "/UK6AMvm_Rz2yA_vR_HFsf0TAM2BLuQalyHZ_xqANkHEG_yKujr/BOkaWamQeL3JYQjN1NUiy_S3J3kw246N_MEmZ0zmmHGj8wP9Ww/no8GoBTJooh2Gv17tyOS1MYXKt0nnkDXRCSYhIMEE4p8iwRfdT/4Wke_KdxtJdRwIk7PMigRKNsaP_Y7ZPrwuaU4CBeGnTHcYR_pl/UxQQ2Bxyo39xPU9rVpV4oDbLWiOqvDxPUavwMFRFp7SYjlpJlT/3RAAU_jqgU4RxR5O4fJ4ij9Y3VrG3YfH2omuhyzcYErjFJ4mCN/b4TAvJUt1BemF4FMwHlb_33TT6GSWeKY_a5fbOm8EZ2f9KZo9t/QT9jAX8IpPhQeBS942ANt8txdypvkHwvZfEeuZdkb0WsJEoael/rTXDwCrXIh6QpLsqVziFdU6ioQ8Y6QxfH0Qt6YuIqdZIE87QWH/cgtk7TSxNu1XTBmMALsoMabFuUaEhgji45IAXtmciSln9x_HPx"}, "attrs": {"temperature_2": {"md": [{"type": "alarm", "name": "very_hot_1", "value": "hot"}, {"type": "alarm", "name": "very_hot_0", "value": "hot"}], "modDate": 1447667724, "type": "celcius", "value": "34", "creDate": 1447667724}, "temperature_0": {"md": [{"type": "alarm", "name": "very_hot_1", "value": "hot"}, {"type": "alarm", "name": "very_hot_0", "value": "hot"}], "modDate": 1447667724, "type": "celcius", "value": "34", "creDate": 1447667724}, "temperature_1": {"md": [{"type": "alarm", "name": "very_hot_1", "value": "hot"}, {"type": "alarm", "name": "very_hot_0", "value": "hot"}], "modDate": 1447667724, "type": "celcius", "value": "34", "creDate": 1447667724}}, "creDate": 1447667724}]
```

##### Delete request
```
DELETE http://localhost:1026/v2/entities/room/attrs/temperature_0
Content-Type: application/json
Fiware-Service: test_delete_service_path
Fiware-ServicePath: /UK6AMvm_Rz2yA_vR_HFsf0TAM2BLuQalyHZ_xqANkHEG_yKujr/BOkaWamQeL3JYQjN1NUiy_S3J3kw246N_MEmZ0zmmHGj8wP9Ww/no8GoBTJooh2Gv17tyOS1MYXKt0nnkDXRCSYhIMEE4p8iwRfdT/4Wke_KdxtJdRwIk7PMigRKNsaP_Y7ZPrwuaU4CBeGnTHcYR_pl/UxQQ2Bxyo39xPU9rVpV4oDbLWiOqvDxPUavwMFRFp7SYjlpJlT/3RAAU_jqgU4RxR5O4fJ4ij9Y3VrG3YfH2omuhyzcYErjFJ4mCN/b4TAvJUt1BemF4FMwHlb_33TT6GSWeKY_a5fbOm8EZ2f9KZo9t/QT9jAX8IpPhQeBS942ANt8txdypvkHwvZfEeuZdkb0WsJEoael/rTXDwCrXIh6QpLsqVziFdU6ioQ8Y6QxfH0Qt6YuIqdZIE87QWH/cgtk7TSxNu1XTBmMALsoMabFuUaEhgji45IAXtmciSln9x_HPx
```

##### Delete response
```
http code: 404
{  
     "error":"NotFound",
     "description":"The requested entity has not been found. Check type and id"
}
```
----------------------------------------------------------------------------------------------------

### Tests features (behave)
```
entities  folder 
     --> list_entities.feature
     --> remove_entity.feature
     --> replace_all_entity_attributes.feature
     --> retrieve_entity.feature
     --> update_existing_entity_attributes.feature
     --> update_or_append_entity_attributes.feature

attributes  folder 
     --> get_attribute_data.feature
     --> update_attribute_data.feature
     --> remove_a_single_attribute.feature


attributes_value folder 
     --> update_attribute_value.feature
```

### PR DESCRIPTION
**Pull request by crbrox**
Fixes #1423

@fgalan 
@kzangeli

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -2,3 +2,5 @@
 - Fix: wrong over-logging at error level updating attributes having metadata without type 
 - Fix: libmicrohttpd 0.9.48 included in contextBroker as static lib (previous Orion versions used 0.9.22 as dynamic library) (Issue #1675)
 - Fix: list of attribute names in URI param 'type' (Issue #1749)
+- Fix: long servicepath when geeting in v2 (#1423)
```
Comment (kzangeli) [inline CHANGES_NEXT_RELEASE]: geeting ... ?

Comment (fgalan) [inline CHANGES_NEXT_RELEASE]: v2 -> NGSIv2

geeting?

Comment (crbrox) [inline CHANGES_NEXT_RELEASE]: Fixed 406029b

File: src/lib/mongoBackend/MongoGlobal.cpp [hunk 2]
```diff
@@ -781,7 +780,7 @@ BSONObj fillQueryServicePath(const std::vector<std::string>& servicePath)
         nullAdded = true;
       }
 
-      char path[SERVICE_NAME_MAX_LEN * 2];
+      char path[SERVICE_PATH_MAX_TOTAL * 2];
       slashEscape(servicePath[ix].c_str(), path, sizeof(path));
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: The change seems to simple ...
From the functest we can see it works but ...

Did we not have a check of service path checking:
- Max total size: 110 (or something like that)
- Max length per 'path-component': 10 (later perhaps changed to 50)
- Max components: 10

What happened to those checks?
I mean, how come they didn't need a modification for this to work?

Comment (kzangeli) [inline src/lib/mongoBackend/MongoGlobal.cpp]: All clarified on skype.
This is simply a bug fix of a lapsus.
The constant SERVICE_PATH_MAX_TOTAL was created but not used. Instead SERVICE_NAME_MAX_LEN was used (which is meant for service, not servicePath).

NTC

**Comment by kzangeli** [comment]
LGTM

**Comment by fgalan** [comment]
LGTM

**Pull request merged**